uipyqt.py

```diff
@@ -0,0 +1,288 @@
+            # remove it from the gui
+            widgetToRemove.setParent(None)
+
+    # this function clears the form from all items in the view
+    # empties all the global collections
+    # and sets all the flags to False
+    def clear_form(self):
+        self.remove_all_widgets(self.opponent_hand_layout)
+        self.opponent_buttons=[]
+
+        self.remove_all_widgets(self.vbox)
+        self.table_buttons=[]
+
+        self.remove_all_widgets(self.my_hand_layout)
+        self.my_hand_buttons=[]
+
+    # sets buttons "Claim cards" and "Lay card" as visible, OK as invisible
+    def enable_button_for_my_move(self):
+        self.claim_cards_button.setVisible(True)
+        self.lay_card_button.setVisible(True)
+        self.OK_button.setVisible(False)
+
+    # sets button OK as visible, "Claim cards" and "Lay card" as invisible
+    def enable_button_for_opponent_move(self):
+        self.claim_cards_button.setVisible(False)
+        self.lay_card_button.setVisible(False)
+        self.OK_button.setVisible(True)
+
+    # draw opponent hand making only the button indicated enabled and checked
+    def draw_opponent_hand(self, card_to_display = ""):
+        if card_to_display!="":
+            displayed_button = QPushButton(card_to_display)
+            displayed_button.setEnabled(True)
+            displayed_button.setCheckable(True)
+            displayed_button.setChecked(True)
+            self.opponent_hand_layout.addWidget(displayed_button)
+            self.opponent_buttons.append(displayed_button)
+
+        for opponent_card in self.sc.hands[1]:
+            opponent_button = QPushButton("XX")
+            opponent_button.setEnabled(False)
+            # add button for each opponent hand
+            self.opponent_hand_layout.addWidget(opponent_button)
+            self.opponent_buttons.append(opponent_button)
+
+    def draw_table(self, disable_cards_not_displayed, cards_to_display=[]):
+
+        # show button for each card on the table
+        for table_card in self.sc.table:
+            table_button = QPushButton(table_card)
+            table_button.setGeometry(200, 150, 100, 40)
+            table_button.setCheckable(True)
+            if disable_cards_not_displayed:
+                table_button.setEnabled(False)
+            table_button.clicked.connect(self.enable_actions)
+            self.vbox.addWidget(table_button)
+            self.table_buttons.append(table_button)
+
+        for displayed_card in cards_to_display:
+            if displayed_card != cards_to_display:
+                displayed_button = QPushButton(displayed_card)
+                displayed_button.setGeometry(200, 150, 100, 40)
+                displayed_button.setEnabled(True)
+                displayed_button.setCheckable(True)
+                displayed_button.setChecked(True)
+                self.vbox.addWidget(displayed_button)
+                self.table_buttons.append(displayed_button)
+
+    def draw_my_hand(self, show_as_active):
```

Comment:
This method and draw_opponent_hand method are very similar. They could be replaced with one method with appropriate parameters.

Reply:
or some common logic could be extracted